### PR TITLE
Switch from API contract version to IsMethodPresent (#6886)

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change adds IsMethodAvailble() to WindowsApiChecker and marks the UniversalApiContractV#_IsAvailble properties as obsolete.

References tot he contract version check in MRTK have been updated to use the new method.

Fixes: #6886

### CODE REVIEW

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealitySpatialMeshObserver.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@
 #endif // UNITY_WSA
 
 #if WINDOWS_UWP
+using Windows.Foundation.Metadata;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealitySpatialMeshObserver.cs]: Is this needed after the move to the MRTK API?

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/Extensions/InteractionSourceExtensions.cs [hunk 2]
```diff
@@ -7,6 +7,7 @@
 using System.Collections.Generic;
 using UnityEngine.XR.WSA.Input;
 using Windows.Foundation;
+using Windows.Foundation.Metadata;
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/Extensions/InteractionSourceExtensions.cs]: Is this needed after the move to the MRTK API?

File: Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs [hunk 3]
```diff
@@ -33,37 +36,64 @@ public static void CheckApiContracts()
...
+        /// <param name="namespaceName">The namespace (ex: "Windows.UI.Input.Spatial") containing the class.</param>
+        /// <param name="className">The name of the class containing the method (ex: "SpatialInteractionMananger").</param>
+        /// <param name="methodName">The name of the method (ex: "IsSourceKindSupported").</param>
+        /// <returns>True if the method is available and can be called. Otherwise, false.</returns>
+        public static bool IsMethodAvailable(
+            string namespaceName,
+            string className,
+            string methodName)
+        {
+#if WINDOWS_UWP
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs]: I think we can call this from the DotNetWinRT APIs too, but that can be added later if needed

File: Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs [hunk 4]
```diff
@@ -33,37 +36,64 @@ public static void CheckApiContracts()
...
+        }
+
+        /// <summary>
+        /// Checks to see if the requested method is present on the current platform.
+        /// </summary>
+        /// <param name="namespaceName">The namespace (ex: "Windows.UI.Input.Spatial") containing the class.</param>
+        /// <param name="className">The name of the class containing the method (ex: "SpatialInteractionMananger").</param>
+        /// <param name="methodName">The name of the method (ex: "IsSourceKindSupported").</param>
+        /// <returns>True if the method is available and can be called. Otherwise, false.</returns>
+        public static bool IsMethodAvailable(
```
Comment (keveleigh) [inline Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs]: We have some similar API calls in MRTK already, like 
https://github.com/microsoft/MixedRealityToolkit-Unity/blob/f060a82bb1c0c9286e372982fdc640ffe6592107/Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealityEyeGazeDataProvider.cs#L65

Doesn't have to be part of this PR, but could be worth adding the `Property` and `Type` versions of this API as well

File: Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealitySpatialMeshObserver.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit.Providers/WindowsMixedReality/XR2018/WindowsMixedRealitySpatialMeshObserver.cs]: good catch!

File: Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs (hunk 4 above)
Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/WindowsApiChecker.cs]: Issue #7426

**Pull request merged**